### PR TITLE
Uncomments out a reproducible build test

### PR DESCRIPTION
**Pull request by nellshamrell**
that appears to now be working on Windows.

I have been investigating (off and on for the past few months) #88982.

In the process, I found that one of the commented out tests does appear to pass now on my Windows workstation. However, the other test does not pass on either Linux or Windows.

Signed-off-by: Nell Shamrell <nells@microsoft.com>

### CODE REVIEW

File: src/test/run-make-fulldeps/reproducible-build/Makefile [hunk 1]
```diff
@@ -12,14 +12,11 @@ all:  \
...
 	remap_cwd_rlib \
 	remap_cwd_to_empty \
+	different_source_dirs_bin \
 	extern_flags
 
-# TODO: Builds of `bin` crate types are not deterministic with debuginfo=2 on
-# Windows.
-# See: https://github.com/rust-lang/rust/pull/87320#issuecomment-920105533
-# Issue: https://github.com/rust-lang/rust/issues/88982
+# This test appears to not be working on either Linux or Windows
```
Comment (nellshamrell) [inline src/test/run-make-fulldeps/reproducible-build/Makefile:18]: To Do: Create a separate issue for this

**Comment by Mark-Simulacrum** [comment]
Try doesn't test all platforms regardless, https://rustc-dev-guide.rust-lang.org/tests/ci.html has some discussion of how to enable the builders in PR CI.

**Comment by nellshamrell** [comment]
Got a failure on one of the Windows builders - investigating.

**Comment by Mark-Simulacrum** [comment]
@rustbot author

`@rustbot ready` to return this to my attention when ready :)

**Comment by nellshamrell** [comment]
Closing this for now - going to re-think my approach and open a new PR when ready.

**Pull request closed without merging**